week05_explore/week5.ipynb

```diff
@@ -413,7 +418,7 @@
     "                action = agent.get_action()\n",
     "                reward = env.pull(action)\n",
     "                agent.update(action, reward)\n",
-    "                scores[agent.name][i] += optimal_reward - reward\n",
+    "                scores[agent.name][i] += optimal_reward - env.action_value(action)\n",
```

Comment:
This is just not want we want to optimize. Here, you are basically taking an expectation over many samples (if you sample `action` many times, your expected reward will be `env._probs[action]`), but the problem statement says that those probabilities are unknown to the agent, and therefore you cannot optimize them directly.

Reply:
Thank you so much for your comment.
The agent does not optimise regret directly since it never observes it, `scores` is only used at `plot_regret` function for visualisation purposes.
`optimal_reward` which equals to `np.max(self._probs)` based on the definition of `env.optimal_reward` function is also an expectation.